presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -68,6 +70,11 @@
 
     private HiveBucketing() {}
 
+    public static int getVirtualBucket(int bucketCount, Path path)
+    {
+        return (hashBytes(0, wrappedBuffer(path.toString().getBytes())) & Integer.MAX_VALUE) % bucketCount;
```

Comment:
Never use `getBytes()` since that uses the platform character set. Instead, use `getBytes(UTF_8)`. However, `Slices` has a shortcut method `utf8Slice()` that takes a `String`.

But taking a step back, why not just do `path.hashCode()`?

Reply:
By using the same hash function in hive bucketing, virtual bucketed table will have the same behavior (e.g. hash distribution) as table that are bucketed on a single `VARCHAR` column. Although I am not sure what the actual benefit keeping them align but I think it's good to be consistent if the extra cost is not that significant.